eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -86,53 +86,93 @@ stages:
                     steps:
                       - checkout: self
 
+                      - pwsh: |
+                          $branchName = "azure-sdk-for-cpp-${{ artifact.Name }}-$(Build.BuildId)"
+                          if ($env:VCPKGFORKBRANCHNAME) { 
+                            Write-Host "Using queue time branch name" 
+                            $branchName = $env:VCPKGFORKBRANCHNAME
+                          } 
+                          Write-Host "##vso[task.setvariable variable=PrBranchName]$branchName"
+                        displayName: Set fork branch name
+
                       - task: Powershell@2
                         inputs:
                           filePath: eng/scripts/Initialize-VcpkgRelease.ps1
                           arguments: >-
-                            -GitHubPat $(azuresdk-github-pat)
                             -GitHubRepo "$(Build.Repository.Name)"
                             -PackageSpecPath $(Pipeline.Workspace)/packages/${{artifact.Name}}/package-info.json
                             -SourceDirectory $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg
                             -Verbose
                           pwsh: true
                         displayName: Initialize vcpkg release
 
+                      # There are potential race conditions if this script runs
+                      # in parallel against the same branch name. Release only 
+                      # one package at a time.
                       - pwsh: |
+                          Write-Host "git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg"
```

Comment:
Wondering..  It might be `nice to have` to make this link an input parameter.  Then we can create draft PRs with, say, my own VCPKG fork for experiments.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1293 tracks this... I think we'll end up needing to do something like this for part of a future CI package testing process.